src/frontend/apps/e2e/__tests__/app-desk/mail-domain.spec.ts

```diff
@@ -0,0 +1,256 @@
+
+const clickOnMailDomainsNavButton = async (page: Page): Promise<void> =>
+  await page.locator('menu').first().getByLabel(`Mail Domains button`).click();
+
+test.describe('Mail domain', () => {
+  test.beforeEach(async ({ page, browserName }) => {
+    await page.goto('/');
+    await keyCloakSignIn(page, browserName);
+  });
+
+  test('redirects to 404 page when the mail domain requested does not exist', async ({
+    page,
+  }) => {
+    await page.route('**/api/v1.0/mail-domains/?page=*', async (route) => {
+      await route.fulfill({
+        json: {
+          count: 0,
+          next: null,
+          previous: null,
+          results: [],
+        },
+      });
+    });
+
+    await page.goto('/mail-domains/unknown-domain.fr');
+    // await expect(page).toHaveURL(/404/, { timeout: 15000 });
+    await expect(
+      page.getByText(
+        'It seems that the page you are looking for does not exist or cannot be displayed correctly.',
+      ),
+    ).toBeVisible({
+      timeout: 15000,
+    });
+  });
+
+  test('checks all the elements are visible when domain exist but contains no mailboxes', async ({
+    page,
+  }) => {
+    const interceptApiCalls = async () => {
+      await page.route(
+        '**/api/v1.0/mail-domains/456ac6ca-0402-4615-8005-69bc1efde43f/mailboxes/?page=1',
+        async (route) => {
+          await route.fulfill({
+            json: {
+              count: 0,
+              next: null,
+              previous: null,
+              results: [],
+            },
+          });
+        },
+      );
+
+      await page.route(
+        '**/api/v1.0/mail-domains/456ac6ca-0402-4615-8005-69bc1efde43f**',
+        async (route) => {
+          await route.fulfill({
+            json: mailDomainDomainFrFixture,
+          });
+        },
+      );
+      await page.route('**/api/v1.0/mail-domains/?page=*', async (route) => {
+        await route.fulfill({
+          json: {
+            count: mailDomainsFixtures.length,
+            next: null,
+            previous: null,
+            results: mailDomainsFixtures,
+          },
+        });
+      });
+    };
+
+    await interceptApiCalls();
+
+    await clickOnMailDomainsNavButton(page);
+
+    await expect(page).toHaveURL(/mail-domains/);
+
+    await page.getByRole('listbox').first().getByText('domain.fr').click();
+    await expect(page).toHaveURL(
+      /mail-domains\/456ac6ca-0402-4615-8005-69bc1efde43f/,
+    );
+
+    await expect(
+      page.getByRole('heading', { name: /domain\.fr/ }).first(),
+    ).toBeVisible();
+
+    await expect(page.getByText('This table is empty')).toBeVisible();
+  });
+
+  test('checks all the elements are visible when domain exists and contains 2 pages of mailboxes', async ({
+    page,
+  }) => {
+    const mailboxesFixtures = {
+      domainFr: {
+        page1: Array.from({ length: 20 }, (_, i) => ({
+          id: `456ac6ca-0402-4615-8005-69bc1efde${i < 10 ? '0' + i : i}f`,
```

Comment:
Why do you complicate your id? I can see you don't have a check on them or I missed something ?